packages/dds/tree/src/test/simple-tree/api/schemaCreationUtilities.spec.ts

```diff
@@ -73,6 +79,47 @@ describe("schemaCreationUtilities", () => {
 		class Parent extends schemaFactory.object("Parent", { mode: Mode.schema }) {}
 	});
 
+	it("enumFromStrings - construction tests", () => {
+		const schemaFactory = new SchemaFactory("com.myApp");
+
+		const ModeNodes = enumFromStrings(new SchemaFactory(`${schemaFactory.scope}.Mode`), [
+			"A",
+			"B",
+			"C",
+		]);
+		type ModeNodes = TreeNodeFromImplicitAllowedTypes<typeof ModeNodes.schema>;
+
+		type I0 = NodeFromSchema<(typeof ModeNodes.schema)[0]>;
+		type I1 = NodeFromSchema<(typeof ModeNodes.schema)[1]>;
+		type I2 = NodeFromSchema<(typeof ModeNodes.schema)[2]>;
+
+		type _checkI = requireTrue<areSafelyAssignable<I0 | I1 | I2, ModeNodes>>;
+
+		type A = NodeFromSchema<typeof ModeNodes.A>;
+		type B = NodeFromSchema<typeof ModeNodes.B>;
+
+		type _checkDifferent = requireFalse<areSafelyAssignable<A, B>>;
```

Comment:
Nit: this will fail if either assignment direction is incompatible, right? Should we test that _each_ direction is not assignable?

Reply:
Yes.

Really this is just checking they are not the same (hence it being called _checkDifferent), since that's the likely failure mode.

You are correct that we could be stricter. This will also pass if either of them is any, which isn't great. Just using isAssignableTo is actually stricter in that stricter in that way, and more clear about what its testing. so I made the change